src/coreclr/src/vm/threadsuspend.cpp

```diff
@@ -3627,224 +3612,141 @@ HRESULT ThreadSuspend::SuspendRuntime(ThreadSuspend::SUSPEND_REASON reason)
             //
             // Suspend the native thread.
             //
-            Thread::SuspendThreadResult str = thread->SuspendThread();
+            Thread::SuspendThreadResult str = thread->SuspendThread(/*fOneTryOnly*/ TRUE);
```

Comment:
Do not loop in suspend on failures. If thread cannot suspend (like if it is forbidding right now), we will deal with other threads and will come back after this one later.